tough/src/editor/targets.rs

```diff
@@ -0,0 +1,556 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#![allow(clippy::used_underscore_binding)] // #20
+
+//! Provides a `TargetsEditor` object for building and editing targets roles.
```

Comment:
How hard would it be to put a usage example in this docstring? Rust will run even run the examples. I don't know if there's way too much setup code required though. https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html